src/acc/acc_bench_smm.c

```diff
@@ -118,20 +116,17 @@ int main(int argc, char* argv[])
 #if defined(_DEBUG)
     fprintf(stderr, "Error: no device found!\n");
 #endif
-#if !defined(__CUDA)
-    CHECK(libsmm_acc_finalize(), NULL);
-#endif
-    CHECK(acc_finalize(), NULL);
```

Comment:
@hfp: this is where I got confused and where I might have undone your fixes for OpenCL?

`c_dbcsr_acc_finalize` is confusingly defined in `cuda/acc_init.cpp`, but the same source file is used for HIP, so I thought the ifdef didn't make sense and removed it. The OpenCL version of this function also call libsmm_acc_finalize, so is this really necessary?

Reply:
We should probably move those shared HIP / CUDA sources out of the cuda/ folder.